src/Command/SynchronizeTaxonCommand.php

```diff
@@ -71,12 +71,25 @@ private function syncTaxonTree(OutputInterface $output): void
     {
         $output->writeln('<info>Sync taxon tree</info>');
 
+        $taxons = [];
         foreach ($this->taxonRepository->findRootNodes() as $rootTaxon) {
             if (!$rootTaxon instanceof TaxonInterface) {
                 continue;
             }
 
-            $this->taxonMessageProducer->synchronize($rootTaxon);
+            $this->extractChildrenFlat($rootTaxon, $taxons);
         }
+
+        $this->taxonMessageProducer->synchronize($taxons);
+    }
+
+    private function extractChildrenFlat(TaxonInterface $rootTaxon, array &$taxons): array
```

Comment:
would not use `&$` use returns and array_merge

Reply:
fixed